### PR TITLE
Update and relax dependencies 

### PR DESCRIPTION
**Pull request by Raphexion**
The main reason for this PR is to make sure that users of this library can use `Poison 5.0.0`. The change of `use Mix.Config` to `import Config` should be OK as the minimal version of this library is Elixir 1.9+.

### CODE REVIEW

**Comment by tanguilp** [comment]
Thanks for the PR! I haven't updated this lib for ages.

Maybe we should make the json lib configurable. Gonna take a look at this, not sure how other libs do it.

File: mix.exs [hunk 1]
```diff
@@ -29,11 +29,11 @@ defmodule JWKSURIUpdater.MixProject do
...
   defp deps do
     [
-      {:dialyxir, "~> 1.0.0-rc.4", only: [:dev], runtime: false},
-      {:ex_doc, "~> 0.19", only: :dev, runtime: false},
+      {:dialyxir, ">= 0.0.0", only: [:dev], runtime: false},
+      {:ex_doc, ">= 0.0.0", only: :dev, runtime: false},
       {:hackney, "~> 1.0", only: :dev},
       {:jose_utils, "~> 0.1"},
-      {:poison, "~> 4.0"},
+      {:poison, "~> 3.0 or ~> 4.0 or ~> 5.0", optional: true},
```
Comment (tanguilp) [inline mix.exs]: We can actually remove the poison dependency. I think this was the default Tesla decoder at some point, but they switched to Jason, which is a dependency of `jose_utils`. So no need for `poison` anymore.

**Review by tanguilp — CHANGES_REQUESTED** [review]

File: mix.exs (hunk 1 above)
Comment (Raphexion) [reply mix.exs]: Great point :sun_with_face: I've removed `:poison` and updated the PR.

**Comment by Raphexion** [comment]
@tanguilp Thank you for writing the library :star:

**Review by tanguilp — APPROVED** [review]

**Comment by tanguilp** [comment]
And thanks for the PR :rocket:

**Comment by tanguilp** [comment]
v1.1.2 published just now

**Pull request merged**